tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -631,7 +634,42 @@ void SPIRVEmitter::doFunctionDecl(const FunctionDecl *decl) {
   curFunction = nullptr;
 }
 
+void SPIRVEmitter::validateVKAttributes(const Decl *decl) {
+  // The frontend will make sure that
+  // * vk::push_constant applies to global variables of struct type
+  // * vk::binding applies to global variables or cbuffers/tbuffers
+  // * vk::counter_binding applies to global variables of RW/Append/Consume
+  //   StructuredBuffer
+  // * vk::location applies to function parameters/returns and struct fields
+  // So the only case we need to check co-existence is vk::push_constant and
+  // vk::binding.
+
+  if (const auto *pcAttr = decl->getAttr<VKPushConstantAttr>()) {
+    if (seenPushConstantAt.isInvalid()) {
+      seenPushConstantAt = pcAttr->getLocation();
+    } else {
+      // TODO: Actually this is slightly incorrect. The Vulkan spec says:
```

Comment:
I don't fully understand this comment. To me it looks like what you're doing here is not incorrect!